src/plugins/condition/components/Condition.vue

```diff
@@ -211,6 +212,50 @@ export default {
             criterionIndex: 0
         };
     },
+    computed: {
+        getSummary: function () {
+            let config = this.domainObject.configuration;
+
+            if (!config.criteria.length) {
+                return 'When all else fails';
+            } else {
+                let rule = '';
+                let summary = 'No criteria specified';
+                if (config.criteria.length === 1 && config.criteria[0].telemetry) {
+                    if (config.criteria[0].operation) {
+                        if (config.criteria[0].input.length ||
+                            (config.criteria[0].operation === 'isDefined' ||
+                             config.criteria[0].operation === 'isUndefined')) {
+                            rule += `When ${config.criteria[0].telemetry.name} value ${this.findDescription(config.criteria[0].operation, config.criteria[0].input)}`
+                            summary = rule;
+                        }
+                    }
+                } else {
+                    let conjunction = '';
+                    summary = config.criteria.length === 1 ? 'No criteria specified' : 'When ';
+                    config.criteria.forEach((criterion, index) => {
+                        rule = '';
+                        if (criterion.operation) {
+                            if (criterion.input.length ||
+                                (criterion.operation === 'isDefined' ||
+                                 criterion.operation === 'isUndefined')) {
+                                rule += `${criterion.telemetry.name} value ${this.findDescription(criterion.operation, criterion.input)}`
+                                if (index === config.criteria.length - 1) {
+                                    conjunction = config.trigger === 'all' ? 'and' : 'or';
+                                } else {
+                                    conjunction = '';
+                                }
+                                summary += ` ${conjunction} ${rule}`
+                            }
+
+                        }
+                    });
+                }
+
+                return summary;
+            }
+        }
+    },
```

Comment:
There are quite a few nested checks here. I might consider using short circuits to clean up a bit. It looks like you can short circuit for the case, `'No criteria specified'`.

Reply:
Got it down to one line now :)